job.js

```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+const fs = require('fs-extra');
+const _ = require('lodash');
+const DummyLogger = require('sq-logger/dummy-logger');
+const Duration = require('sq-toolkit/duration');
+
+const template = require('./lib/utils/template');
+const BigQueryError = require('./error');
+const BigQueryFactory = require('./factory');
+
+const COST_THRESHOLD_IN_GB = 100;
+
+class BigQueryJob {
+
+    /**
+     * @param {Object}    opts
+     * @param {String=}   opts.name
+     * @param {String}    opts.sqlFilename
+     * @param {Number=}   [opts.costThresholdInGB = 100]
+     *
+     * @param {BigQuery=} opts.bigQuery
+     * @param {WinstonLogger|DummyLogger=} opts.logger
+     */
+    constructor(opts) {
+        this.name = opts.name || null;
+        this.sqlFilename = opts.sqlFilename || null;
+        this.costThresholdInGB = opts.costThresholdInGB || COST_THRESHOLD_IN_GB;
+
+        this.bigQuery = opts.bigQuery || null ;
+        this.logger = opts.logger || new DummyLogger();
+
+        this.sqlStr = null;
+        this.sqlTemplate = null;
+        this._initialized = false;
+    }
+
+    /**
+     * @return {Boolean}
+     */
+    isInitialized() {
+        return this._initialized === true;
+    }
+
+    /**
+     * @return {Promise<BigQueryJob>}
+     */
+    async init() {
+        if (this.isInitialized() === true) {
+            return this;
+        }
+        this.bigQuery = this.bigQuery || BigQueryFactory.getInstance();
+        this.sqlStr = await fs.readFile(this.sqlFilename, 'utf8');
+        this.sqlTemplate = template(this.sqlStr);
+        this._initialized = true;
+        return this;
+    }
+
+    /**
+     * @return {Object|null}
+     */
+    getQueryParams() {
+        return null;
+    }
+
+    /**
+     * @return {String}
+     */
+    getQuerySQLDebug() {
+        return '';
+    }
+
+    /**
+     * @return {String}
+     */
+    getQuerySQL() {
+        const debugStr = this.getQuerySQLDebug();
+        const params = this.getQueryParams();
+        return (debugStr ? (debugStr.trim() + '\n') : '') + this.sqlTemplate(params || {});
+    }
+
+    /**
+     * @params {Object=}  opts
+     * @params {Boolean=} opts.dryRun
+     * @params {Boolean=} opts.useLegacySql
+     * @params {String=}  opts.query
+     * @return {Object}
+     */
+    getQueryOptions(opts) {
+        return {
+            // This params CAN be overridden
+            dryRun: null,
```

Comment:
I see you override this function with a boolean value. I don't know what happens with a null value. Shouldn't it be a boolean value, for example false as default?